### PR TITLE
List projects and facts about them

### PR DESCRIPTION
**Pull request by fdegir**
This commit creates initial lists of projects and documents facts
about them. The list of projects and their facts may contain errors
and they are expected to be corrected via subsequent commits or PRs.

### Applicable Issues
#6 

### Description of the Change
This change creates initial list of projects hosted by Eiffel Community on GitHub together with the project facts such as description of the project, link to the project main repository, and the lifecycle stage the project resides.

### Alternate Designs
The community can continue as is, handling everything hosted within Eiffel Community Organization as individual repositories.

### Benefits
Having a document listing the projects together with their facts is beneficial for both current Eiffel Community members and the people who may be interested in using and contributing to Eiffel as it gives people one place to get a high level understanding of what projects are available for them. 

In addition to providing visibility, it also makes it possible for users to see which projects are in which stage in their lifecycle so the users can take conscious decisions when it comes to using one or more projects for evaluation and/or production purposes. (See [this issue](https://github.com/eiffel-community/eiffel-repository-template/issues/9) as an example)

### Possible Drawbacks
It is possible that the list of projects and the facts about them getting out of sync but if the community follows the [project lifecycle](https://github.com/eiffel-community/community/blob/master/PROJECT_LIFECYCLE.md) process while making changes to the existing projects or creating new ones, this risk can be mitigated.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Signed-off-by: Fatih Degirmenci fatih.degirmenci@est.tech

### CODE REVIEW

File: PROJECTS.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| Eiffel Commons          | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-commons)          | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Easy2Use         | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-easy2use)         | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Event Repository | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-event-repository) | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Gerrit Herald    | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-gerrit-herald)    | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel GraphQL API      | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-graphql-api)      | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
```
Comment (t-persson) [inline PROJECTS.md:40]:
```suggestion
| Eiffel GraphQL API      | A GraphQL API-layer set between the eiffel message bus and a MongoDB | [Link](https://github.com/eiffel-community/eiffel-graphql-api)      | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
```

File: PROJECTS.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+| Eiffel Intelligence     | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-intelligence)     | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Jenkins Plugin   | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-jenkins-plugin)   | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Python Library   | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-pythonlib)        | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
```
Comment (t-persson) [inline PROJECTS.md:43]:
```suggestion
| Eiffel Python Library   | A python library for subscribing to and publishing Eiffel events to a message-broker. | [Link](https://github.com/eiffel-community/eiffel-pythonlib)        | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
```

**Comment by fdegir** [comment]
this was the first time i used the commit suggestions feature which is cool. Thanks @t-persson for giving me chance to try this. 😃 

but i have no idea about how i can bring the latest commits back to my clone in case if i need to send new commits to incorporate other comments this PR may receive. any pointer helps.

**Comment by fdegir** [comment]
> but i have no idea about how i can bring the latest commits back to my clone in case if i need to send new commits to incorporate other comments this PR may receive. any pointer helps.

nevermind; it seems the commit from accepted suggestions also go to my fork as well so i can pull that branch to my clone and get the commit.

**Review by e-backmark-ericsson — CHANGES_REQUESTED** [review]
I think the column with links to main repo could be replaced with having a link on the project name in the first column.

Also, the descriptions could be taken directly from the About section on the respective main repository.

According to the definitions of the Graduated stage (e.g. maintainers from multiple organizations) the only project that has reached that stage is the protocol itself. So I'd like to see that the protocol was displayed as Graduated.

**Comment by fdegir** [comment]
> I think the column with links to main repo could be replaced with having a link on the project name in the first column.

will update the table. we can perhaps add link to the maintainer teams in that column instead.

> Also, the descriptions could be taken directly from the About section on the respective main repository.

ok!

> According to the definitions of the Graduated stage (e.g. maintainers from multiple organizations) the only project that has reached that stage is the protocol itself. So I'd like to see that the protocol was displayed as Graduated.

Yeah, I was thinking about that as well but the reason why I put protocol to Sandbox stage is to apply the project lifecycle process to protocol as the first thing so the created issue/pr to move the protocol to Graduated can be an example to other projects.

File: PROJECTS.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+| Eiffel Protocol         | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel)                  | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Commons          | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-commons)          | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Easy2Use         | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-easy2use)         | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Event Repository | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-event-repository) | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
+| Eiffel Gerrit Herald    | TBD                                                             | [Link](https://github.com/eiffel-community/eiffel-gerrit-herald)    | [Sandbox](./PROJECT_LIFECYCLE.md#stage-sandbox) |
```
Comment (m-linner-ericsson) [inline PROJECTS.md:39]: Was there a reason not to include:

- eiffel-gerrit-plugin https://github.com/eiffel-community/eiffel-gerrit-plugin
- eiffel-jira-plugin https://github.com/eiffel-community/eiffel-jira-plugin

File: PROJECTS.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+This document lists existing projects, links to the main project repository, and the lifecycle stage the
+projects currently reside in.
+
+Any change to the project facts listed in this document shall be done according to the instructions documented
+in [project lifecycle process](./PROJECT_LIFECYCLE.md).
+
+## Project Facts
+
+The table below lists the projects hosted by Eiffel Community alphabetically. (except the protocol itself)
```
Comment (m-linner-ericsson) [inline PROJECTS.md:28]: Does the the following wording make it more clear that "except the protocol itself" is about the sorting?

```suggestion
The table below lists the projects hosted by Eiffel Community alphabetically with the protocol itself listed first.
```

File: PROJECTS.md (hunk 3 above)
Comment (fdegir) [reply PROJECTS.md:39]: nope. i probably missed while browsing the repositories in GitHub Eiffel Community organization. Thanks for the pointer, will fix.

**Comment by fdegir** [comment]
I went through all the project main repositories to get descriptions and found few issues listed below.

Projects that lack description
- https://github.com/eiffel-community/eiffel-commons
- https://github.com/eiffel-community/eiffel-jira-plugin

Projects with description but better to be added by others
- https://github.com/eiffel-community/eiffel-easy2use
- https://github.com/eiffel-community/eiffel-sepia

I am also wondering if community should be added as a project as well or not.

**Comment by e-backmark-ericsson** [comment]
It could be listed for completeness, so that all main repos and maintainer mail lists are linked from this document, but with 'Not Applicable' on the lifecycle status.
You can keep the missing descriptions empty to be updated by future PRs

**Review by e-backmark-ericsson — APPROVED** [review]
LGTM

**Review by m-linner-ericsson — APPROVED** [review]

**Review by sodero — APPROVED** [review]
LGTM++

**Pull request merged**